### ISSUE TITLE
FxA-102: geoprofile login attempts

### ISSUE THREAD

**Issue by rfk** [mozilla/fxa-features#23]

**Comment by davismtl** [comment]
Feature doc: https://docs.google.com/document/d/1jspHtYks7j5ikd_uAtO9IB4Lh7emWOdrmuDniTebuAA/edit

**Comment by vladikoff** [comment]
Added some notes to the Google doc

**Comment by vladikoff** [comment]
from mtg:
- might have a prototype
- feature doc still in the works

**Comment by davismtl** [comment]
Question from legal: "If I visit a new city, will that city start be whitelisted forever?" 
What will the rules be around that? (not in doc yet)
If we haven't thought of it yet, what would we like to do?

**Comment by TDA** [comment]
@davismtl I believe @jvehent had a solution for that when new cities keep getting added, you split them into `hotzones` kinda? With a ranking for frequency of logins?
Either that, or I completely imagined that.

**Comment by jvehent** [comment]
>  If we haven't thought of it yet, what would we like to do?

@rfk mentioned only keeping the last 6 connections, so the new one would remove the oldest in the history.

If we want to keep a lifetime geoprofile, we need to calculate a trusted geoffence for each user that increases up to a certain size. Then new connections inside the geoffence reinforce its trust, while new connections outside the geoffence that are verified create a new geoffence for the user.
That prevents having to keep exact connection history (you just keep a geocenter and a radius per geoffence) but is significantly more difficult to implement.

**Comment by vbudhram** [comment]
> If I visit a new city, will that city start be whitelisted forever?

No, you would not be whitelisted forever. The feature is being implemented by only evaluating the last x login attempts. Also, something to note, even if the login event occurs in a similar area, there are other rules that might get triggered and prompt the user for verification, ie has it seen this user-agent before.

**Comment by rfk** [comment]
We could also explore a more explicitly time-based criteria, like "you logged in from this geo area in the last 6 months".

More broadly through, and perhaps as a follow-up feature, I think it would make sense for us to show the login activity to the user and given them the option of discarding it.

**Comment by davismtl** [comment]
Thanks for all the answers. This makes sense. I think last X amount of cities + time-based criteria is a good combination.

**Comment by davismtl** [comment]
from mtg:
- @vbudhram is doing some analysis on this. Getting stats first and then analysis to determine how many people this will impact.

**Comment by vbudhram** [comment]
Analysis Notes: https://bugzilla.mozilla.org/show_bug.cgi?id=1332712
